packages/headless/src/features/commerce/context/context-slice.ts

```diff
@@ -0,0 +1,47 @@
+import {createReducer} from '@reduxjs/toolkit';
+import {getContextInitialState} from './context-state';
+import {setClientId, setContext, setCurrency, setLanguage, setTrackingId, setUser, setView} from './context-actions';
+
+export const contextReducer = createReducer(
+  getContextInitialState(),
+
+  (builder) => {
+    builder
+      .addCase(setContext, (state, {payload}) => {
+        if (payload.trackingId) {
```

Comment:
This reducer does not really behave like I would expect a `setFoo` to behave, ie: overwrite the full context, even undefined properties.

It seems it would be impossible, with the actions that are defined here, to remove any values.

Should it just be :

```typescript
 .addCase(setContext, (state, {payload}) => {
       state = payload;
}